File: src/main/java/com/dansoftware/boomega/plugin/api/ThemePlugin.java
package com.dansoftware.boomega.plugin.api;

import com.dansoftware.boomega.gui.theme.Theme;
import com.dansoftware.boomega.gui.theme.ThemeMeta;
import org.jetbrains.annotations.NotNull;

/**
 * Allows to add a custom theme to the application.
 *
 * @author Daniel Gyoerffy
 */
public interface ThemePlugin extends BoomegaPlugin {
    @NotNull
    ThemeMeta<? extends Theme> getThemeMeta();
    @NotNull
    Theme getTheme();
}

File: src/main/java/com/dansoftware/boomega/plugin/api/ThemePlugin.kt
package com.dansoftware.boomega.plugin.api

import com.dansoftware.boomega.gui.theme.Theme

/**
 * Allows to add a custom theme to the application.
 *
 * @author Daniel Gyoerffy
 */
interface ThemePlugin : BoomegaPlugin {
    val theme: Theme
}